plugins/async-node/jvm/src/test/kotlin/com/intuit/playerui/plugins/asyncnode/AsyncNodePluginTest.kt

```diff
@@ -228,4 +228,64 @@ internal class AsyncNodePluginTest : PlayerTest() {
         }
         Assertions.assertTrue(true)
     }
+
+    @TestTemplate
+    fun `handle multiple updates through callback mechanism`() = runBlockingTest {
+        var update: Asset? = null
+        var updateCount = 0
+
+        plugin?.hooks?.onAsyncNode?.tap("") { _, node, callback ->
+            when (updateCount) {
```

Comment:
Hmm I don't think this makes sense from a functional POV.

This tap is for the original resolve, where you are given the async node and a callback to be stored for reference later in case an update is needed.

The code here actually uses the callback to update the original node within the "first" tap, which means line 252 doesn't ever get hit (that return value is supposed to be the resolve for the original async node)

Furthermore, updateCount 0 conceptually doesn't click with me either, because the updateCount should be 1 even before the async node is resolved (from first tree update before it's resolved by the consumer)

You can reference the core test [here](https://github.com/player-ui/player/pull/404/files#diff-fbc47a3e301bdf595a6d686ded538fe6767e534af8821c421f97dc56264d1332R146) to see how the update was intended to be used